src/Models/Feature.php

```diff
@@ -4,25 +4,43 @@
 
 namespace JustSteveKing\Laravel\FeatureFlags\Models;
 
+use Carbon\Carbon;
+use Exception;
 use Illuminate\Database\Eloquent\Model;
 use Illuminate\Database\Eloquent\Relations\BelongsToMany;
+use Illuminate\Support\Facades\App;
 use JustSteveKing\Laravel\FeatureFlags\Models\Concerns\NormaliseName;
 use JustSteveKing\Laravel\FeatureFlags\Models\Builders\FeatureBuilder;
 
 class Feature extends Model
 {
     use NormaliseName;
-    
+
     protected $fillable = [
         'name',
         'description',
         'active',
+        'expires_at'
     ];
 
     protected $casts = [
         'active' => 'boolean',
+        'expires_at' => 'datetime'
     ];
 
+    public static function booted()
+    {
+        static::retrieved(function(Feature $feature) {
+            if(
```

Comment:
Again, more of a stylistic thing, but I find this section a bit difficult to read with all the nested conditionals and ternary expressions. I think this would benefit from a couple of intermediate variables.

Reply:
I'm all for clean code - but nested conditionals make this harder to read and test, which means it could cause issues further down the line

Reply:
Thank you both, I've refactored this now :)